### PR TITLE
abi: define debug abi version for user space dbg interfaces

### PR DESCRIPTION
**Pull request by mmaka1**
The sof-logger and potentially other debug API clients perform ABI
compatibility check using the single FW ABI version. The same one is used
by the primary FW client which is the kernel driver. If there is a change
made to the debug API, the main ABI has to be updated to protect integrity
of the debug tools while such a change may not affect the kernel driver
at all.

This patch introduces new debug ABI version to be increased when changing
user space debug interfaces while the the main ABI is not affected.
Recompilation and installation of the new driver every time the tunneled
debug protocol is upgraded may be avoided.

Signed-off-by: Marcin Maka <marcin.maka@linux.intel.com>

### CODE REVIEW

**Comment by mmaka1** [comment]
Implements #2374

File: rimage/file_simple.c [hunk 1]
```diff
@@ -72,6 +73,73 @@ static int get_mem_zone_type(struct image *image, Elf32_Shdr *section)
 	return SOF_FW_BLK_TYPE_INVALID;
 }
 
+static int fw_version_copy(struct snd_sof_logs_header *header,
```
Comment (paulstelian97) [inline rimage/file_simple.c]: I feel like the separating out of this function can be its own patch.

**Comment by lgirdwood** [comment]
Do we need corresponding kernel PR for this one ?

**Comment by mmaka1** [comment]
> Do we need corresponding kernel PR for this one ?

@lgirdwood I tested with unchanged kernel and seems to work fine. Another fw_ready extension does not break compatibility, therefore I increased 'minor' component only in the main ABI before aligning initial debug ABI version with the main one.

**Comment by lgirdwood** [comment]
> > Do we need corresponding kernel PR for this one ?
> 
> @lgirdwood I tested with unchanged kernel and seems to work fine. Another fw_ready extension does not break compatibility, therefore I increased 'minor' component only in the main ABI before aligning initial debug ABI version with the main one.

Ok great, @plbossart good for you ? or would you need a patch that printk() the ABI version for userspace debug ?

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@zrombel @xiulipan CI is showing a pass afaict but we are seeing pending/failed ?

**Comment by zrombel** [comment]
Quickbuild changes are causing some problems, sorry for that. When it will be fixed I will rebuild this PR.

**Comment by zrombel** [comment]
QB is back on track. Build and tests passed.

File: rimage/file_simple.c (hunk 1 above)
Comment (jajanusz) [reply rimage/file_simple.c]: If it were just refactor, then yea. But here you'd have 1 diff to have it changed in another.

File: src/include/user/abi_dbg.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+#include <kernel/abi.h>
+
+#ifndef __USER_ABI_DBG_H__
+#define __USER_ABI_DBG_H__
+
+#define SOF_ABI_DBG_MAJOR 3
+#define SOF_ABI_DBG_MINOR 14
+#define SOF_ABI_DBG_PATCH 0
```
Comment (jajanusz) [inline src/include/user/abi_dbg.h]: Is 'debug abi' the best name for this? We have folders that are named by-abi-user (kernelspace or userspace), but ABI version symbols are named by-feature (generic vs debug).

Comment (jajanusz) [reply src/include/user/abi_dbg.h]: Nvm, I see it's going to be more granular for user, versions are in `sof_ipc_user_abi_version `

**Comment by jajanusz** [comment]
`sof-ci/jenkins/pr-tools-build Pending — Tools Building` shows as pending, but it's CI reporting error, cos results there are skipped/ok

**Comment by lgirdwood** [comment]
@mmaka1 could you copy the rimage patches to the external repo to as another PR.
@fredoh9 @xiulipan any idea why tools build is timing out ? Server OK ?

**Comment by jajanusz** [comment]
@mmaka1 can u fix checkpatch issue?

**Comment by xiulipan** [comment]
@lgirdwood As we are trying to update the GitHub behind a proyx, there is always some chance that the status update is missed. Will try to add some more strict check to see if we can use retry or other method to avoid such issue.

**Comment by mmaka1** [comment]
@jajanusz I fixed the alignment issue. Another one be fixed is the `memcpy` issue however this function is used in many places in that source file, so there is a general replacement, done as another patch, required.

**Comment by lgirdwood** [comment]
@zrombel we have a DMIC CI failure on ICL, but it should be unrelated to this patch ?

**Comment by wwittbrx** [comment]
@lgirdwood Aside from SSP KD tests failing due to regression on master (should be fixed by https://github.com/thesofproject/sof/pull/2440) this looks good to be merged.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@wwittbrx running CI again now that Jenkins has had some fixes too.
